### PR TITLE
emacs: add v29.4

### PR DESCRIPTION
**Pull request by alecbcs**
Add emacs v29.4.

**Changelog:**
https://git.savannah.gnu.org/cgit/emacs.git/tree/etc/NEWS?h=emacs-29

### CODE REVIEW

File: var/spack/repos/builtin/packages/emacs/package.py [hunk 1]
```diff
@@ -15,12 +15,15 @@ class Emacs(AutotoolsPackage, GNUMirrorPackage):
     homepage = "https://www.gnu.org/software/emacs"
     git = "git://git.savannah.gnu.org/emacs.git"
     gnu_mirror_path = "emacs/emacs-24.5.tar.gz"
+    list_url = " https://ftpmirror.gnu.org/emacs/"
+    list_depth = 0
 
     maintainers("alecbcs")
 
     license("GPL-3.0-or-later")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/emacs/package.py]: I did double-check this :-)

```suggestion
    license("GPL-3.0-or-later", checked_by="wdconinc")
```

**Review by tldahlgren — DISMISSED** [review]
Confirmed sha256 BUT getting `All fetchers failed for spack-stage-emacs-master-rppxetqtvik6lnyxtqq3znfmavg6lwxj` for master.

**Comment by tldahlgren** [comment]
@alecbcs Let me know what you want to do about `master`.

**Comment by wdconinc** [comment]
I think this works:
```diff
-git = "git://git.savannah.gnu.org/emacs.git"
+git = "https://git.savannah.gnu.org/git/emacs.git"
```

**Comment by alecbcs** [comment]
@tldahlgren I'm not seeing any issues with master on my end. Does your script assume `https://` instead of `git://` ?

**Comment by alecbcs** [comment]
@wdconinc `git.savannah.gnu.org` doesn't appear to support https clones in my testing.

**Comment by wdconinc** [comment]
@alecbcs It's what they advertise for anonymous clones at https://savannah.gnu.org/git/?group=emacs. Only member clones are direct to git://.

**Comment by wdconinc** [comment]
> @wdconinc `git.savannah.gnu.org` doesn't appear to support https clones in my testing.

```
$ git clone https://git.savannah.gnu.org/git/emacs.git
Cloning into 'emacs'...
remote: Counting objects: 1120477, done.
remote: Compressing objects: 100% (208500/208500), done.
```

**Comment by alecbcs** [comment]
Ah! I see the URL has to be updated to `/git/emacs`.

**Comment by tldahlgren** [comment]
> @tldahlgren I'm not seeing any issues with master on my end. Does your script assume `https://` instead of `git://` ?

My script didn't work so I used `spack fetch`, which is how I confirmed the error message.

**Review by tldahlgren — APPROVED** [review]
I can fetch `master now`. Thanks.

**Pull request merged**